cmd/util/ledger/migrations/legacy_controller_migration.go

```diff
@@ -0,0 +1,59 @@
+package migrations
+
+import (
+	"bytes"
+	"encoding/hex"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/ledger"
+)
+
+// LegacyControllerMigration migrates previous registers
+// that were using the legacy controller value (value is always set to the same owner value).
+// the controller value should now all be empty and
+// in future can be remove all together.
+type LegacyControllerMigration struct {
+	Logger zerolog.Logger
+}
+
+func (lc *LegacyControllerMigration) Migrate(payload []ledger.Payload) ([]ledger.Payload, error) {
+	for _, p := range payload {
+		owner := p.Key.KeyParts[0].Value
+		controller := p.Key.KeyParts[1].Value
+		key := p.Key.KeyParts[2].Value
+
+		if len(controller) > 0 {
+			if bytes.Equal(owner, controller) {
+				// case - public key count
+				if string(key) == state.KeyPublicKeyCount {
+					p.Key.KeyParts[1].Value = []byte("")
+					continue
+				}
+				// case - public keys
+				if bytes.HasPrefix(key, []byte("public_key_")) {
+					p.Key.KeyParts[1].Value = []byte("")
+					continue
+				}
+				// case - contract names
+				if string(key) == state.KeyContractNames {
+					p.Key.KeyParts[1].Value = []byte("")
+					continue
+				}
+				// case - contracts
+				if bytes.HasPrefix(key, []byte(state.KeyCode)) {
+					p.Key.KeyParts[1].Value = []byte("")
+					continue
```

Comment:
The code in all blocks is the same, maybe combine the cases.
Also, do the keys need to be allocated byte slices? Can they just be `nil`?

```suggestion
				if string(key) == state.KeyPublicKeyCount ||
					bytes.HasPrefix(key, []byte("public_key_")) ||
					string(key) == state.KeyContractNames ||
					bytes.HasPrefix(key, []byte(state.KeyCode)) {
					
					p.Key.KeyParts[1].Value = nil
					continue
```

Reply:
I didn't know if we need allocation or nil would work but using an empty slice was the way FVM was setting those values, so I tried to prevent accidental issues in this PR, anyway we are going to kill that controller field soon.

Reply:
updated the conditions to reuse the code.